examples/codeviewer/common/src/jvmMain/kotlin/org/jetbrains/codeviewer/platform/JvmFile.kt

```diff
@@ -59,12 +59,14 @@ fun java.io.File.toProjectFile(): File = object : File {
 
             override fun get(index: Int): String {
                 val startPosition = lineStartPositions[index]
-                val length = if (index + 1 < size) lineStartPositions[index + 1] - startPosition else
-                    byteBufferSize - startPosition
+                var endPosition = if (index + 1 < size) lineStartPositions[index + 1] - 1 else byteBufferSize
+                if (endPosition < byteBufferSize && byteBuffer[endPosition].toInt().toChar() == '\r') {
```

Comment:
What about files that don't contain `\r` symbol, just `\n`?

Reply:
I see that we substract: `- 1`. We should probably do that for `byteBufferSize` as well. Or better check for `\n` to avoid assumptions that the last symbol is `\n`